willow-eventhandler/src/main/java/com/nitorcreations/willow/eventhandler/EventPoller.java

```diff
@@ -0,0 +1,231 @@
+package com.nitorcreations.willow.eventhandler;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.nitorcreations.willow.messages.event.EventMessage;
+import com.nitorcreations.willow.messages.metrics.MetricConfig;
+import com.nitorcreations.willow.sshagentauth.SSHUtil;
+
+/**
+ * Event poller for receiving events from metrics server via websocket and dispatching them to event handlers.
+ * 
+ * @author mtommila
+ */
+public class EventPoller {
+  private Logger logger = Logger.getLogger(getClass().getName());
+
+  private Map<String, List<EventHandler>> eventHandlers;
+
+  @Inject
+  private Gson gson;
+
+  @Inject
+  private ExecutorService executorService;
+
+  /**
+   * Initialize the event poller.
+   * 
+   * @param eventHandlers The handlers for different event types.
+   * @param uri The web socket connection URI.
+   */
+  public void initialize(Map<String, List<EventHandler>> eventHandlers, URI uri) {
+    this.eventHandlers = eventHandlers;
+    EventListener listener = new EventListener(uri);
+    executorService.submit(listener);
+    logger.fine("Started event listener");
+  }
+
+  /**
+   * Websocket message listener. Dispatches events to the event handlers.
+   * Autowired to Jetty websocket implementation via annotations.
+   * 
+   * @author mtommila
+   */
+  @WebSocket
+  public class EventListener implements Runnable {
+
+    private Logger logger = Logger.getLogger(getClass().getName());
+
+    private URI uri;
+    private WebSocketClient client;
+    private CountDownLatch connectLatch;
+    private Set<String> messageIds = new HashSet<>();
+
+    private EventListener(URI uri) {
+      this.uri = uri;
+    }
+
+    private synchronized WebSocketClient getClient() {
```

Comment:
getClient vs createClient is a bit confusing. I propose to delete the getClient and make the two callers (run and closeClient handle a null client instaed).